sound/soc/sof/core.c

```diff
@@ -227,12 +227,10 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 	sdev->first_boot = false;
 
 	/*
-	 * Register audio client.
-	 * This can fail but errors cannot be propagated.
+	 * register audio client devices.
+	 * This could fail but errors cannot be propagated
```

Comment:
Generic comment on this patch:
I don't think it's completely reasonable to require a split between DMIC, SSP and HDaudio in different clients.
The moment you want to enable Acoustic echo cancellation, you'll be in a world of hurt, with the echo reference handled by a client but needed by another. Another example is ASRC between DMIC and SSP slave which Seppo is working on.

It's really only HDMI that is completely orthogonal to the other links

In other words, there are cases where we might want to split DMIC/SSP, and we should allow for modular designs, but we should not make that a requirement. splitting is fine is there is not interaction between links, it's not it they do interact.

Reply:
> In other words, there are cases where we might want to split DMIC/SSP, and we should allow for modular designs, but we should not make that a requirement. splitting is fine is there is not interaction between links, it's not it they do interact.

@plbossart just trying to understand the requirement: is this interaction within the FW with the pipelines from different clients interacting with each other?

Can we possibly think of handling such a case with a virtual endpoint between the 2 topologies? I am not sure how I would do this. But in my mind,  isnt the scenario similar to what happens in the case of a a guest OS trying to play/record audio by interacting with the pipelines in the FW?

Reply:
Separating just the HDMI pipelines into a separate topology should be doable. I just need to figure out the partitioning in the HDA probe sequence. Let me think about this.

Reply:
> > In other words, there are cases where we might want to split DMIC/SSP, and we should allow for modular designs, but we should not make that a requirement. splitting is fine is there is not interaction between links, it's not it they do interact.
> 
> @plbossart just trying to understand the requirement: is this interaction within the FW with the pipelines from different clients interacting with each other?
> 
> Can we possibly think of handling such a case with a virtual endpoint between the 2 topologies? I am not sure how I would do this. But in my mind, isnt the scenario similar to what happens in the case of a a guest OS trying to play/record audio by interacting with the pipelines in the FW?

If each client enables a 'card', then it's my understanding that the DAPM parts are limited to what is visible in each client. We'd need to define 'proxies' or 'gateways' between topologies, it'd be painful to design.